### PR TITLE
[Stability] Cancel polling for old votes

### LINKED ISSUE #2222: [Stability] - Fix poll cancelling for votes
### What is this task and why do we need to work on it?

Right now we don't cancel polls properly for votes; this causes us to continue polling for old views 0..n that have already become irrelevant.

### What work will need to be done to complete this task?

Add poll cancelling for votes when we create a certificate. Also add it for timeouts.

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

- [ ] a manual run where no old polls are still present

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by rob-maron**
Closes #2222 

### This PR: 

Fixes a bug wherein we are not cancelling old polls for votes.

Cancels polls for votes in the following places:
- when we receive a timeout in view sync or a finalized view sync cert 
- when we form a QC or DAC 
- when we time out for both the quorum and view sync (DA already had this)

Also cancels a poll for proposals for view 1, as this is covered by the latest proposal mechanism. For a node finishing catchup, it would constantly try to pull the first proposal.

### This PR does not: 

Add any GC improvements/more advanced proposal logic/cancelling as discussed. This will be in another PR.

### Key places to review: 

`da.rs`, `consensus.rs`, `view_sync.rs` to make sure all the cancels are in a good spot.

### CODE REVIEW

File: crates/task-impls/src/view_sync.rs
Comment (bfish713) [inline crates/task-impls/src/view_sync.rs:791]: Same comment we want `CancelPollForViewSyncVotes`

**Review by bfish713 — CHANGES_REQUESTED** [review]

File: crates/task-impls/src/view_sync.rs
Comment (bfish713) [reply crates/task-impls/src/view_sync.rs:791]: Also I think we should `CancelPollForViewSyncCertificate` here as well

Comment (rob-maron) [reply crates/task-impls/src/view_sync.rs:791]: You're right 🙈

Comment (rob-maron) [reply crates/task-impls/src/view_sync.rs:791]: Should we also cancel the certificate poll in the case of a timeout?

Comment (bfish713) [reply crates/task-impls/src/view_sync.rs:793]: Which certificate and which timeout?

Comment (rob-maron) [reply crates/task-impls/src/view_sync.rs:793]: In the case of a `Timeout` event in the view sync task, should we cancel polls for the certificate of that view ?

Comment (bfish713) [reply crates/task-impls/src/view_sync.rs:793]: `Timeout` is the start of view sync (if it's the second in a row) so probably shouldn't cancel any view sync related things

File: crates/task-impls/src/view_sync.rs
Comment (bfish713) [inline crates/task-impls/src/view_sync.rs:470]: I think it was correct before, we just timed out consensus for the round and are now maybe entering view sync, we should cancel polling for quorum votes and proposals but that should be handled in the consensus task

File: crates/task-impls/src/consensus.rs
Comment (bfish713) [inline crates/task-impls/src/consensus.rs:995]: Cancel poll for proposal here too, I think

**Review by bfish713 — APPROVED** [review]
LGTM, Can you wait for CI to pass before trying to merge

**Pull request merged**
